src/library/scala/collection/Iterable.scala

```diff
@@ -549,21 +549,21 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
     *
     */
   def groupBy[K](f: A => K): immutable.Map[K, C] = {
-    val m = mutable.Map.empty[K, Builder[A, C]]
+    val m = immutable.HashMap.newBuilder[K, Builder[A, C]].asInstanceOf[immutable.HashMapBuilder[K, Builder[A, C]]]
```

Comment:
Since you have relaxed the visibility of class `HashMapBuilder` to the `collection` package, you could instantiate it and avoid the cast?